osu.Game.Tests/Chat/MessageFormatterTests.cs

```diff
@@ -273,6 +273,66 @@ public void TestMarkdownFormatLinkWithEscapedAndBalancedBrackets()
             Assert.AreEqual(21, result.Links[0].Length);
         }
 
+        [Test]
+        public void TestMarkdownFormatLinkWithInlineTitle()
+        {
+            Message result = MessageFormatter.FormatMessage(new Message { Content = "I haven't seen [this link format](https://osu.ppy.sh \"osu!\") before..." });
+
+            Assert.AreEqual("I haven't seen this link format before...", result.DisplayContent);
+            Assert.AreEqual(1, result.Links.Count);
+            Assert.AreEqual("https://osu.ppy.sh", result.Links[0].Url);
+            Assert.AreEqual(15, result.Links[0].Index);
+            Assert.AreEqual(16, result.Links[0].Length);
+        }
+
+        [Test]
+        public void TestMarkdownFormatLinkWithInlineTitleAndEscapedQuotes()
+        {
+            Message result = MessageFormatter.FormatMessage(new Message { Content = "I haven't seen [this link format](https://osu.ppy.sh \"inner quote \\\" just to confuse \") before..." });
+
+            Assert.AreEqual("I haven't seen this link format before...", result.DisplayContent);
+            Assert.AreEqual(1, result.Links.Count);
+            Assert.AreEqual("https://osu.ppy.sh", result.Links[0].Url);
+            Assert.AreEqual(15, result.Links[0].Index);
+            Assert.AreEqual(16, result.Links[0].Length);
+        }
+
+        [Test]
+        public void TestMarkdownFormatLinkWithUrlInTextAndInlineTitle()
```

Comment:
Also what does the test do? If it fails, if it works, please rename the test so you can see what it should do.

Example: ifAThanReturnSomething 

The test name must make clear what is tested and how

Reply:
same goes to every other Testname

Reply:
Again, I am familiar with that convention and sometimes use it, but choose not to here due to pre-existing conventions. The value of such naming is disputable to me, and anyone lacking context can check out the commit log and PR description.